third_party/2and3/chardet/universaldetector.pyi

```diff
@@ -1,19 +1,39 @@
-from typing import Dict, Union, AnyStr, Pattern
+from typing import Dict, Union, AnyStr, Pattern, Optional
 from logging import Logger
 
-class UniversalDetector:
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+
+    class FinalResultType(TypedDict):
+        encoding: str
+        confidence: float
+        language: str
+
+    class IntermediateResultType(TypedDict):
```

Comment:
This too.

Reply:
And I had a doubt. If I have a set of files all of which use a specific `TypedDict` class for typing. Should I include the `TypedDict` class in all these files or is there a way I can define it in one place and import it into those files?

Reply:
You can just import it from another file.

Reply:
Just like python modules? Cool.

Reply:
@JelleZijlstra Are there any conventions to be followed for the`.pyi`file where the `TypedDict` class is defined? The definition which can then be imported to other files?

Reply:
The name should probably be prefixed with `_` to indicate that it doesn't exist at runtime.

Reply:
Got another doubt. Can you help me out?

I came across a line like this:

    if sys.version_info < (3, 0):
        base_str = (str, unicode)
    else:
        base_str = (bytes, str)

What should its type be? I figure it starts with something like `Tuple[Type]` but couldn't be sure. What do you think?

Reply:
@JelleZijlstra I think I found another way to keep the common class definition in one place. Just defined it inside the `__init__.pyi` file. Can then load with `from . import <type>`.